ggml/src/ggml-sycl.cpp

```diff
@@ -3581,7 +3581,8 @@ static void ggml_sycl_mul_mat(ggml_backend_sycl_context & ctx, const ggml_tensor
 
     bool use_mul_mat_vec_q =  ggml_is_quantized(src0->type)
         && src1->type == GGML_TYPE_F32 && dst->type == GGML_TYPE_F32
-        && src1->ne[1] <= MMVQ_MAX_BATCH_SIZE;
+        && src1->ne[1] <= MMVQ_MAX_BATCH_SIZE
+        && ctx.stream()->get_backend() == sycl::backend::ext_oneapi_cuda ? true: src1->ne[1] > MMVQ_MIN_BATCH_SIZE;
```

Comment:
I've performed some testing and it seems that the change has introduced an error in the MUL_MAT dispatch. The f32 to f32 case is trying to go through the MMVQ codepath, see the test backend log:

```
MUL_MAT(type_a=f32,type_b=f32,m=16,n=16,k=256,bs=[1,1],nr=[1,1]):
/builds/llama.cpp/ggml/src/ggml-sycl/mmvq.cpp:1020: fatal error
```
Which is (as expected) a runtime exception thrown due to the invalid types. I think the issue relates to operator precedence and the following suggestion should fix it:

```suggestion
        && (ctx.stream()->get_backend() == sycl::backend::ext_oneapi_cuda || src1->ne[1] > MMVQ_MIN_BATCH_SIZE);
```